doc/man3/EVP_PKEY_get_default_digest_nid.pod

```diff
@@ -30,6 +30,50 @@ are keys with a B<EVP_PKEY_ASN1_METHOD>; these keys have typically
 been loaded from engines, or created with L<EVP_PKEY_assign_RSA(3)> or
 similar.
 
+=head1 PARAMETERS
+
+EVP_PKEY_get_default_digest_name() uses these <OSSL_PARAM(3)> key to ask the
+provider for the values it seeks:
+
+=over 4
+
+=item "mandatory-digest" (B<OSSL_PKEY_PARAM_MANDATORY_DIGEST>) <UTF8 string>
+
+When the providers responds to this, the return value from
+EVP_PKEY_get_default_digest_name() will be 2.
+
+If the value from the provider is C<""> or C<"UNDEF">,
+EVP_PKEY_get_default_digest_name() will place the string C<"UNDEF"> into
+I<mdname>.  This means that no digest should be used with the signature
+operation.
+
+=item "default-digest" (B<OSSL_PKEY_PARAM_DEFAULT_DIGEST>) <UTF8 string>
```

Comment:
The description of the meaning of these parameters seems more useful to provider authors rather than callers of the functions covered by this man page. I would expect these descriptions to instead be in the doc/man7/provider-keymgmt.pod file. Provider authors would almost certainly not think to look on this page.

Reply:
There seems to be different view on that.  You'll find similar documentation in EVP_DigestInit.pod, for example.

However, nothing stops me from *also* adding documentation in doc/man7/provider-keymgmt.pod (and possibly refer to it from this file)

Reply:
I'll do a bit of shuffling

Reply:
Shuffling done, please have a look

Reply:
I see little point in it being in this file. It is of no relevance to the callers of this function and provider authors won't look here. The `EVP_DigestInit.pod` file is different because that page documents functions that can be used by callers to query specific parameters - so its more justifiable there. That isn't the case here.

Reply:
Okie